airflow/jobs/scheduler_job.py

```diff
@@ -289,7 +290,12 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
             .filter(not_(DM.is_paused))
             .filter(TI.state == TaskInstanceState.SCHEDULED)
             .options(selectinload('dag_model'))
-            .order_by(-TI.priority_weight, DR.execution_date)
+            .order_by(
+                TI.priority_weight.desc(),
+                desc(case([(TI.last_scheduling_decision.is_(None), 1)], else_=0)),
+                TI.last_scheduling_decision.desc(),
```

Comment:
Please use `airflow.utils.sqlalchemy.nulls_first` here if we can.

It might not work out of the box for `DESC` sorted column.

Reply:
The nulls_first didn't work for other DBs except postgres. 

I'm preferring this PR https://github.com/apache/airflow/pull/19747 over this. What do you think?

Reply:
Used nullsfirst for postgresql. I used it on the column directly instead of using `airflow.utils.sqlalchemy.nulls_first` since we are sure it's postgres